src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIQuarrier.java

```diff
@@ -139,14 +141,55 @@ public IAIState loadRequirements()
                 return IDLE;
             }
 
-            final WorkOrderMiner wo = new WorkOrderMiner(quarry.getStructurePack(), "infrastructure/mineshafts/" + quarry.getSchematicName() + "shaft1.blueprint", quarry.getSchematicName() + "shaft1", quarry.getRotation(), quarry.getPosition().below(2), false, building.getPosition());
+            final Tuple<String, String> shaft = getShaftPath(quarry);
+            final WorkOrderMiner wo = new WorkOrderMiner(quarry.getStructurePack(), shaft.getA(), shaft.getB(), quarry.getRotation(), quarry.getPosition().below(2), false, building.getPosition());
             building.getColony().getWorkManager().addWorkOrder(wo, false);
             job.setWorkOrder(wo);
         }
 
         return super.loadRequirements();
     }
 
+    private Tuple<String, String> getShaftPath(@NotNull final IBuilding quarry)
+    {
+        String path = "infrastructure/mineshafts/" + quarry.getSchematicName() + "shaft1.blueprint";
+
+        final AbstractTileEntityColonyBuilding tileEntity = quarry.getTileEntity();
+        if (tileEntity != null)
+        {
+            path = quarry.getBlueprintPath().replace('\\', '/')
```

Comment:
are those seperators the same on all OS's? or should we be using the Files.seperator thing instead

Reply:
Windows will produce backslashes but will accept both, which is one reason this converts strictly to forward slashes.

Linux will produce forward slashes and only accepts forward slashes.

MacOS classic used a different separator but MacOS BSD also uses forward slashes.

So forward slashes are safe everywhere, and it makes life easier when the tag can assume a single format.